File: module/com.sandpolis.core.profile/src/main/java/com/sandpolis/core/profile/store/Events.java
//============================================================================//
//                                                                            //
//                Copyright © 2015 - 2020 Subterranean Security               //
//                                                                            //
//  This source file is subject to the terms of the Mozilla Public License    //
//  version 2. You may not use this file except in compliance with the MPL    //
//  as published by the Mozilla Foundation at:                                //
//                                                                            //
//    https://mozilla.org/MPL/2.0                                             //
//                                                                            //
//=========================================================S A N D P O L I S==//
package com.sandpolis.core.profile.store;

import com.sandpolis.core.instance.event.ParameterizedEvent;

public final class Events {

	public static final class ProfileOnlineEvent extends ParameterizedEvent<Profile> {
	}

	public static final class ProfileOfflineEvent extends ParameterizedEvent<Profile> {
	}

	private Events() {
	}
}

File: instance/sandpolis-client/common/src/main/kotlin/org/s7s/instance/client/desktop/Events.kt
//============================================================================//
//                                                                            //
//            Copyright © 2015 - 2022 Sandpolis Software Foundation           //
//                                                                            //
//  This source file is subject to the terms of the Mozilla Public License    //
//  version 2. You may not use this file except in compliance with the MPLv2. //
//                                                                            //
//============================================================================//
package org.s7s.instance.client.desktop.ui

import javafx.scene.control.TitledPane
import tornadofx.*

object Events {

    class MainViewChangeEvent(val view: String) : FXEvent()

    class MainMenuOpenEvent(val view: TitledPane) : FXEvent()
}
